panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system/robot_driver/roboteq_data_converters.hpp

```diff
@@ -219,24 +226,31 @@ class RoboteqData
 
   const FaultFlag & GetFaultFlag() const { return fault_flags_; }
   const ScriptFlag & GetScriptFlag() const { return script_flags_; }
-  const RuntimeError & GetLeftRuntimeError() const { return left_runtime_error_; }
-  const RuntimeError & GetRightRuntimeError() const { return right_runtime_error_; }
+
+  /**
+   * @brief Returns runtime error flags for the given channel
+   *
+   * @param channel 1 or 2
+   * @return runtime error flags
+   * @throws std::runtime_error if invalid channel number
+   */
+  const RuntimeError & GetRuntimeError(const std::uint8_t channel) const;
 
   std::string GetFlagErrorLog() const;
 
   std::map<std::string, bool> GetFlagErrorMap() const;
   std::map<std::string, bool> GetErrorMap() const;
 
 private:
-  MotorState left_motor_state_;
-  MotorState right_motor_state_;
+  MotorState channel_1_motor_state_;
+  MotorState channel_2_motor_state_;
```

Comment:
What about static arrays for states and errors?
```suggestion
 std::array<MotorState, 2> motor_states_;
```

Reply:
This may completely change when refactoring data converters so will leave it as it is.